### PR TITLE
Up to date library to properly work with gestpay

### PR DESCRIPTION
**Pull request by marcio199226**
-Added missing response parameters
-Fixed SoapClient because I have an issue with user_agent if not set under http key, fixed verify_peer & ssl

I have used it on our company's project so it works like a charm

### CODE REVIEW

File: src/WSCryptDecryptSoapClient.php [hunk 1]
```diff
@@ -33,7 +33,6 @@ class WSCryptDecryptSoapClient
     public function __construct($testEnv = false, $caFile = null)
     {
         $soapClientDefaultOption = array(
-            'user_agent' => 'EndelWar-GestPayWS/1.3 (+https://github.com/endelwar/GestPayWS)',
```
Comment (endelwar) [inline src/WSCryptDecryptSoapClient.php]: Why removing user agent from default soap client options and moving it to the bottom?

File: src/WSCryptDecryptSoapClient.php [hunk 2]
```diff
@@ -83,9 +82,10 @@ private function getStreamContext($testEnv = false, $caFile = null)
             $host = $this->certificatePeerName['production'];
         }
 
-        $this->streamContextOption['ssl']['crypto_method'] = STREAM_CRYPTO_METHOD_TLS_CLIENT;
-        $this->streamContextOption['ssl']['verify_peer'] = true;
-        $this->streamContextOption['ssl']['SNI_enabled'] = true;
+        $this->streamContextOption['ssl']['crypto_method'] = STREAM_CRYPTO_METHOD_ANY_CLIENT;
+        $this->streamContextOption['ssl']['allow_self_signed'] = true;
+        $this->streamContextOption['ssl']['verify_peer'] = false;
+        $this->streamContextOption['ssl']['SNI_enabled'] = false;
```
Comment (endelwar) [inline src/WSCryptDecryptSoapClient.php]: you are removing all security check on certificate validation, why?

File: src/WSCryptDecryptSoapClient.php [hunk 3]
```diff
@@ -83,9 +82,10 @@ private function getStreamContext($testEnv = false, $caFile = null)
             $host = $this->certificatePeerName['production'];
         }
 
-        $this->streamContextOption['ssl']['crypto_method'] = STREAM_CRYPTO_METHOD_TLS_CLIENT;
-        $this->streamContextOption['ssl']['verify_peer'] = true;
-        $this->streamContextOption['ssl']['SNI_enabled'] = true;
+        $this->streamContextOption['ssl']['crypto_method'] = STREAM_CRYPTO_METHOD_ANY_CLIENT;
```
Comment (endelwar) [inline src/WSCryptDecryptSoapClient.php]: Using any client permit lowering crypto method down to sslv2, Gestpay is moving everything to tlsv1.2
(also STREAM_CRYPTO_METHOD_ANY_CLIENT is availably from php 5.6, library minimum requirement is php 5.3.3)

File: src/WSCryptDecryptSoapClient.php [hunk 4]
```diff
@@ -104,9 +104,9 @@ private function getStreamContext($testEnv = false, $caFile = null)
             }
         } else {
             $this->streamContextOption['ssl']['peer_name'] = $host;
-            $this->streamContextOption['ssl']['verify_peer_name'] = true;
+            $this->streamContextOption['ssl']['verify_peer_name'] = false;
```
Comment (endelwar) [inline src/WSCryptDecryptSoapClient.php]: also here peer name must be validated to be sure to send money to the real Gestpay server :)

**Review by endelwar — CHANGES_REQUESTED** [review]

**Comment by endelwar** [comment]
Thanks for your submission!

> Fixed SoapClient because I have an issue with user_agent if not set under http key

Which error are you experiencing? I've this code on more that 100 ecommerce and it's working non-stop

**Comment by marcio199226** [comment]
@endelwar i have experienced an 
`SOAP-ERROR:` Parsing WSDL: Couldn't load from....`

On 1.3.0v, anyway fixing error above i have another issue with response from gestpay, now we have 4 new params and they are missing in your DecryptResponse.

Anyway you are right disabling all this security checks isn't a good practice.

I encoutered the first error on php 5.6.x and on 7.0.22
I have openssl ext installed (output from php -m) and self signed certificate

I forced to use STREAM_CRYPTO_METHOD_TLSv1_2_CLIENT above php 5.6.7 because since 5.6.7 this constant STREAM_CRYPTO_METHOD_TLS_CLIENT is equal to STREAM_CRYPTO_METHOD_TLSv1_0_CLIENT and this cause the error.

Anyway reverting all options (verify_peer & SNI_enabled)  except verify_peer_name to the previous values it works, but with verify_peer_name = true i have an error: 
`Could not connect to host`

Ok i got it you had the old url for test env, it was https://testecomm.sella in your case which returns 307 and I'm not sure that verify_peer_name can handle this, so I updated it to https://sandbox.gestpay.net and respectively for peer name to sandbox.gestpay.net

I hope now you can accept & merge my pull request

**Comment by endelwar** [comment]
> Ok i got it you had the old url for test env, it was https://testecomm.sella in your case which returns 307 and I'm not sure that verify_peer_name can handle this, so I updated it to https://sandbox.gestpay.net and respectively for peer name to sandbox.gestpay.net

Yes, that's the problem! Gestpay changed test url, so I didn't notice any problems on production...

I'll be glad to merge your PR, just fix the user-agent string (try first with default soap options) before

**Review by endelwar — APPROVED** [review]

**Pull request merged**